httpx_cache/cache/redis.py

```diff
@@ -0,0 +1,102 @@
+import threading
+import typing as tp
+from abc import ABC
+
+import anyio
+import httpx
+from httpx_cache.cache.base import BaseCache
+from httpx_cache.serializer.base import BaseSerializer
+from httpx_cache.serializer.common import MsgPackSerializer
+from httpx_cache.utils import get_cache_key
+from redis import Redis
+from redis.asyncio import Redis as AsyncRedis
+
+
+class RedisCache(BaseCache, ABC):
+    lock = threading.Lock()
+
+    def __init__(
+        self, serializer: tp.Optional[BaseSerializer] = None, **configs
+    ) -> None:
+        self.cache = Redis(**configs)
```

Comment:
I'm more in favor of using `Redis.from_url(..)` here, since it's a more straightforward interface and provides a `ConnectionPool` directly: https://redis.readthedocs.io/en/latest/connections.html#redis.Redis.from_url